# Support of country and its subdivision and script.
diff --git a/node_modules/@mmomtchev/expo-android-localized-app-name/index.js b/node_modules/@mmomtchev/expo-android-localized-app-name/index.js
index b42ab21..e07225a 100644
--- a/node_modules/@mmomtchev/expo-android-localized-app-name/index.js
+++ b/node_modules/@mmomtchev/expo-android-localized-app-name/index.js
@@ -22,9 +22,15 @@ function withAndroidLocalizedName(config) {
                         resources.push({ string: { $: { name: key }, _: strings[key] } });
                 }
                 if (resources.length) {
-                    await fs.promises.mkdir(path.resolve(resPath, `values-${locale}`), { recursive: true });
+                    let finalLocale = locale;
+                    if (finalLocale.includes("-")) {
+                        finalLocale = `b+${finalLocale.replaceAll("-", "+")}`;
+                    }
+                    await fs.promises.mkdir(path.resolve(resPath, `values-${finalLocale}`), {
+                        recursive: true,
+                    });
                     await fs.promises.writeFile(
-                        path.resolve(resPath, `values-${locale}`, 'strings.xml'),
+                        path.resolve(resPath, `values-${finalLocale}`, 'strings.xml'),
                         builder.buildObject({ resources })
                     );
                 }
