# Fix react-native-segmented-control/segmented-control#667.
diff --git a/node_modules/@react-native-segmented-control/segmented-control/index.d.ts b/node_modules/@react-native-segmented-control/segmented-control/index.d.ts
index b914477..040f29c 100644
--- a/node_modules/@react-native-segmented-control/segmented-control/index.d.ts
+++ b/node_modules/@react-native-segmented-control/segmented-control/index.d.ts
@@ -6,11 +6,12 @@ import {
   ViewProps,
   NativeSyntheticEvent,
   NativeMethods,
-  Constructor,
   TargetedEvent,
   ViewStyle,
 } from 'react-native';
 
+type Constructor<T> = new (...args: any[]) => T;
+
 export interface NativeSegmentedControlIOSChangeEvent extends TargetedEvent {
   value: string;
   selectedSegmentIndex: number;
