diff --git a/node_modules/native-base/lib/typescript/components/primitives/Button/types.d.ts b/node_modules/native-base/lib/typescript/components/primitives/Button/types.d.ts
index aec4dd2..dacbca6 100644
--- a/node_modules/native-base/lib/typescript/components/primitives/Button/types.d.ts
+++ b/node_modules/native-base/lib/typescript/components/primitives/Button/types.d.ts
@@ -8,6 +8,7 @@ import type { ISpinnerProps } from '../Spinner/types';
 import type { IIconProps } from '../Icon';
 import type { InterfacePressableProps } from '../Pressable/types';
 import type { ColorSchemeType } from '../../../components/types';
+import { IBoxProps } from 'native-base/src/components/primitives';
 export interface InterfaceButtonProps extends InterfacePressableProps<IButtonProps> {
     /**
      * The color of the radio when it's checked. This should be one of the color keys in the theme (e.g."green", "red").
@@ -71,6 +72,10 @@ export interface InterfaceButtonProps extends InterfacePressableProps<IButtonPro
      * Props to be passed to the HStack used inside of Button.
      */
     _stack?: Partial<IStackProps>;
+    /**
+     * Props to be passed to the Box used inside of Button.
+     */
+    _box?: Partial<IBoxProps>;
     /**
      * Props to be passed to the Icon used inside of Button.
      */
diff --git a/node_modules/native-base/src/components/primitives/Button/Button.tsx b/node_modules/native-base/src/components/primitives/Button/Button.tsx
index 705efd0..3d59a5e 100644
--- a/node_modules/native-base/src/components/primitives/Button/Button.tsx
+++ b/node_modules/native-base/src/components/primitives/Button/Button.tsx
@@ -48,6 +48,7 @@ const Button = (
     onBlur,
     _text,
     _stack,
+    _box,
     _spinner,
     isLoadingText,
     _icon,
@@ -104,7 +105,7 @@ const Button = (
   );
 
   const boxChildren = (child: any) => {
-    return child ? <Box _text={_text}>{child}</Box> : null;
+    return child ? <Box _text={_text} {..._box}>{child}</Box> : null;
   };
 
   return (
diff --git a/node_modules/native-base/src/components/primitives/Button/types.ts b/node_modules/native-base/src/components/primitives/Button/types.ts
index 2aa215f..fe9601e 100644
--- a/node_modules/native-base/src/components/primitives/Button/types.ts
+++ b/node_modules/native-base/src/components/primitives/Button/types.ts
@@ -14,6 +14,7 @@ import type { ISpinnerProps } from '../Spinner/types';
 import type { IIconProps } from '../Icon';
 import type { InterfacePressableProps } from '../Pressable/types';
 import type { ColorSchemeType } from '../../../components/types';
+import { IBoxProps } from 'native-base';
 
 // const myFunction = ({ a, b }) => {
 //   return { a: a, b: b };
@@ -94,6 +95,10 @@ export interface InterfaceButtonProps
    * Props to be passed to the HStack used inside of Button.
    */
   _stack?: Partial<IStackProps>;
+  /**
+   * Props to be passed to the Box used inside of Button.
+   */
+  _box?: Partial<IBoxProps>;
   /**
    * Props to be passed to the Icon used inside of Button.
    */
