diff --git a/node_modules/splatnet3-types/dist/generated-type-helpers.d.ts b/node_modules/splatnet3-types/dist/generated-type-helpers.d.ts
index a6001cc..c448e28 100644
--- a/node_modules/splatnet3-types/dist/generated-type-helpers.d.ts
+++ b/node_modules/splatnet3-types/dist/generated-type-helpers.d.ts
@@ -1,4 +1,4 @@
 import { request_types as OperationFields } from './request-types.js';
-export declare type LinkedFieldType<ParentType extends {}, TypeName extends string & keyof ParentType, Type, Plural extends boolean> = Plural extends true ? Type[] : null extends ParentType[TypeName] ? Type | null : Type;
+export declare type LinkedFieldType<ParentType extends {}, TypeName extends string & keyof ParentType, Type, Plural extends boolean> = Plural extends true ? null[] extends ParentType[TypeName] ? (Type | null)[] : Type[] : null extends ParentType[TypeName] ? Type | null : Type;
 export declare type OperationFieldType<ParentTypeName extends string, TypeName extends string, Type> = ParentTypeName extends keyof OperationFields ? TypeName extends keyof OperationFields[ParentTypeName] ? UnknownScalarType extends Type ? OperationFields[ParentTypeName][TypeName] : null extends OperationFields[ParentTypeName][TypeName] ? Type | null : Type : Type : Type;
 export declare type UnknownScalarType = unknown & (string | number | boolean);
diff --git a/node_modules/splatnet3-types/dist/splatoon3ink.d.ts b/node_modules/splatnet3-types/dist/splatoon3ink.d.ts
index ac30c0a..b433138 100644
--- a/node_modules/splatnet3-types/dist/splatoon3ink.d.ts
+++ b/node_modules/splatnet3-types/dist/splatoon3ink.d.ts
@@ -23,9 +23,9 @@ declare type CoopGroupingSchedule_splatoon3ink = Omit<StageScheduleQuery_730cd98
     bigRunSchedules: Pick<Connection<CoopSchedule_splatoon3ink>, 'nodes'>;
 };
 declare type CoopSchedule_splatoon3ink = Omit<CoopSchedule_schedule, 'setting'> & {
-    setting: CoopSetting_splatoon3ink;
+    setting: CoopSetting_splatoon3ink | null;
 };
-declare type CoopSetting_splatoon3ink = Omit<CoopSetting_schedule, 'weapons'> & {
+declare type CoopSetting_splatoon3ink = Omit<NotNullable<CoopSetting_schedule>, 'weapons'> & {
     weapons: CoopSupplyWeapon_splatoon3ink[];
 };
 declare type CoopSupplyWeapon_splatoon3ink = Pick<CoopSupplyWeapon, 'name' | 'image'> & {
diff --git a/node_modules/splatnet3-types/dist/types.d.ts b/node_modules/splatnet3-types/dist/types.d.ts
index 70103b3..4e26b51 100644
--- a/node_modules/splatnet3-types/dist/types.d.ts
+++ b/node_modules/splatnet3-types/dist/types.d.ts
@@ -95,7 +95,7 @@ export interface PhotoAlbumItem {
     uploadedTime: string;
 }
 export interface Catalog {
-    progress: CatalogProgress;
+    progress: CatalogProgress | null;
     seasonName: string;
     seasonEndTime: string;
     bonus: CatalogBonus;
@@ -270,8 +270,8 @@ export interface CoopHistoryDetail {
     jobBonus: number | null;
 }
 export interface CoopBossResult {
-    boss: unknown;
-    hasDefeatBoss: unknown;
+    boss: CoopEnemy;
+    hasDefeatBoss: boolean;
 }
 export interface CoopPlayerResult {
     player: CoopPlayer;
